docs/migration/index.md

```diff
@@ -0,0 +1,378 @@
+)
+
+var (
+  //...
+  ModuleBasics = module.NewBasicManager(
+    //...
+    slashing.AppModuleBasic{},
+    // Add feegrantmodule.AppModuleBasic{},
+    feegrantmodule.AppModuleBasic{}, // <--
+    ibc.AppModuleBasic{},
+    //...
+  )
+  //...
+)
+
+type App struct {
+  //...
+  // Replace codec.Marshaler with codec.Codec
+  appCodec          codec.Codec // <--
+  // Add FeeGrantKeeper
+  FeeGrantKeeper   feegrantkeeper.Keeper // <--
+}
+
+func New(...) {
+  //bApp.SetAppVersion(version.Version)
+  bApp.SetVersion(version.Version) // <--
+
+  keys := sdk.NewKVStoreKeys(
+    //...
+    upgradetypes.StoreKey,
+    // Add feegrant.StoreKey
+    feegrant.StoreKey, // <--
+    evidencetypes.StoreKey,
+    //...
+  )
+
+  app.FeeGrantKeeper = feegrantkeeper.NewKeeper(appCodec, keys[feegrant.StoreKey], app.AccountKeeper)  // <--
+  // Add app.BaseApp as the last argument to upgradekeeper.NewKeeper
+  app.UpgradeKeeper = upgradekeeper.NewKeeper(skipUpgradeHeights, keys[upgradetypes.StoreKey], appCodec, homePath, app.BaseApp)
+  
+  app.IBCKeeper = ibckeeper.NewKeeper(
+    // Add app.UpgradeKeeper
+    appCodec, keys[ibchost.StoreKey], app.GetSubspace(ibchost.ModuleName), app.StakingKeeper, app.UpgradeKeeper, scopedIBCKeeper,
+  )
+
+  govRouter.AddRoute(govtypes.RouterKey, govtypes.ProposalHandler).
+    //...
+    // Replace NewClientUpdateProposalHandler with NewClientProposalHandler
+    AddRoute(ibchost.RouterKey, ibcclient.NewClientProposalHandler(app.IBCKeeper.ClientKeeper))
+
+  // Replace porttypes with ibcporttypes
+  ibcRouter := ibcporttypes.NewRouter()
+
+  app.mm.SetOrderBeginBlockers(
+    upgradetypes.ModuleName,
+    // Add capabilitytypes.ModuleName,
+    capabilitytypes.ModuleName,
+    minttypes.ModuleName,
+    //...
+    // Add feegrant.ModuleName,
+    feegrant.ModuleName,
+  )
+
+  // Add app.appCodec as an argument to module.NewConfigurator:
+  app.mm.RegisterServices(module.NewConfigurator(app.appCodec, app.MsgServiceRouter(), app.GRPCQueryRouter()))
+  
+  // Replace:
+  // app.SetAnteHandler(
+  // 	ante.NewAnteHandler(
+  // 		app.AccountKeeper, app.BankKeeper, ante.DefaultSigVerificationGasConsumer,
+  // 		encodingConfig.TxConfig.SignModeHandler(),
+  // 	),
+  // )
+
+  // With the following:
+  anteHandler, err := ante.NewAnteHandler(
+    ante.HandlerOptions{
+      AccountKeeper:   app.AccountKeeper,
+      BankKeeper:      app.BankKeeper,
+      SignModeHandler: encodingConfig.TxConfig.SignModeHandler(),
+      FeegrantKeeper:  app.FeeGrantKeeper,
+      SigGasConsumer:  ante.DefaultSigVerificationGasConsumer,
+    },
+  )
+  if err != nil {
+    panic(err)
+  }
+  app.SetAnteHandler(anteHandler)
+
+  // Remove the following:
+  // ctx := app.BaseApp.NewUncachedContext(true, tmproto.Header{})
+  // app.CapabilityKeeper.InitializeAndSeal(ctx)
+}
+
+func (app *App) InitChainer(ctx sdk.Context, req abci.RequestInitChain) abci.ResponseInitChain {
+  var genesisState GenesisState
+  // Replace tmjson with json
```

Comment:
```suggestion
```

I should have specified this in my previous suggestion. It was `tmjson` before, we changed this to using `json` package initially but it was a mistake, this must remain `tmjson` and therefore there are no change in this part

Reply:
Got it, thanks.